# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1472120624 14400
#      Thu Aug 25 06:23:44 2016 -0400
# Node ID ddac3d15ad7d08d8a49a9dbbf64ac77fae28ae7d
# Parent  805442f0674b7f092a6825c698a4b0d82b03f4a5
use octave_refcount class for reference counts

* input.h (octave_base_reader::count): Use octave_refcount object.
* thread-manager.h (octave_base_thread_manager::count): Likewise.

diff --git a/libgui/src/thread-manager.h b/libgui/src/thread-manager.h
--- a/libgui/src/thread-manager.h
+++ b/libgui/src/thread-manager.h
@@ -18,16 +18,20 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if ! defined (octave_thread_manager_h)
 #define octave_thread_manager_h 1
 
+#include "octave-config.h"
+
+#include "oct-refcount.h"
+
 class octave_base_thread_manager
 {
 public:
 
   friend class octave_thread_manager;
 
   octave_base_thread_manager (void) : count (1) { }
 
@@ -38,17 +42,17 @@ public:
   virtual ~octave_base_thread_manager (void) { }
 
   virtual void register_current_thread (void) = 0;
 
   virtual void interrupt (void) = 0;
 
 protected:
 
-  int count;
+  octave_refcount<int> count;
 };
 
 class octave_thread_manager
 {
 public:
 
   octave_thread_manager (void);
 
diff --git a/libinterp/corefcn/input.h b/libinterp/corefcn/input.h
--- a/libinterp/corefcn/input.h
+++ b/libinterp/corefcn/input.h
@@ -26,16 +26,17 @@ along with Octave; see the file COPYING.
 #define octave_input_h 1
 
 #include "octave-config.h"
 
 #include <cstdio>
 
 #include <string>
 
+#include "oct-refcount.h"
 #include "oct-time.h"
 #include "ovl.h"
 #include "pager.h"
 
 class octave_value;
 namespace octave
 {
   class base_lexer;
@@ -132,17 +133,17 @@ public:
   virtual bool input_from_terminal (void) const { return false; }
 
   virtual bool input_from_file (void) const { return false; }
 
   virtual bool input_from_eval_string (void) const { return false; }
 
 private:
 
-  int count;
+  octave_refcount<int> count;
 
   int pflag;
 
   octave::base_lexer *lexer;
 
   void do_input_echo (const std::string&) const;
 
   static const std::string in_src;
